### ISSUE TITLE
Allowing raw limits

### ISSUE THREAD

**Issue by Linksku** [knex/knex#4805]
# Environment

Knex version: 0.95.11
Database + version: Materialize 0.9.10 (based on Postgres)
OS: Ubuntu 20

# Feature discussion / request

1. Explain what is your use case

Materialize (https://github.com/MaterializeInc/materialize) doesn't allow prepared values for LIMIT. As far as I can tell, Knex always makes the LIMIT value a prepared value. The only way to hardcode it in the query is something like:

```
knex('foo')
  .orderByRaw(`1 limit ${limit}`);
```

Would it be possible to have a raw limit value?

2. Explain what kind of feature would support this

Allow `.limit(raw(10))`.

3. Give some API proposal, how the feature should work

Make `.limit()` accept `raw()` or add `.limitRaw()`

**Comment by Linksku** [comment]
Closing because this works:

```
Knex.QueryBuilder.extend('limitRaw', function (sql: string, bindings: any[]) {
  // @ts-ignore
  this._statements.push({
    grouping: 'limit',
    type: 'limitRaw',
    value: this.client.raw(sql, bindings),
  });
  return this;
});
```

**Comment by kibertoad** [comment]
I think we should provide a better solution out of the box.
@OlivierCavadenti thoughts?

**Comment by OlivierCavadenti** [comment]
I see you expose the problem here : https://github.com/MaterializeInc/materialize/issues/8961
I see the problem well, but as I can see with 'raw' functions is to inject piece of SQL in the query.
joinRaw for example is useful to make natural joins.
But with 'limit' I cant see examples where you can inject more than just a number ?

By the way, to address the problem I see more an optional boolean.

**Comment by kibertoad** [comment]
I agree, optional param seems to be the way to go.

**Comment by OlivierCavadenti** [comment]
@Linksku a parameter like could be ok ? (parameter name need to be better defined I think):
``` javascript
knex('foo')
  .orderBy('1')
  .limit(1, { raw: true });
```
Could be nice to make that for offset function too.

**Comment by Linksku** [comment]
Yep that would work! Though maybe something related to "binding" or "param" would be clearer, like "skipBinding".

**Comment by OlivierCavadenti** [comment]
@Linksku I made a PR where you can make this two syntaxes :

``` javascript
knex('foo')
  .orderBy('1')
  .limit(1, { skipBinding: true })
  .offset(2, { skipBinding: true });
```
or shorter:
``` javascript
knex('foo')
  .orderBy('1')
  .limit(1, true)
  .offset(2, true);
```

Feel free to post comments on the PR

**Comment by kibertoad** [comment]
Released in 1.0.1.

**Comment by sebastianoblom** [comment]
I ran into this problem today.
SphinxQL, which is a dialect for SphinxSearch ( fulltext search server - http://sphinxsearch.com/ ) uses for format `offset, limit` instead of `limit 1 offset 2` when using limits.

The only way I have found using knex qb around this problem is to append raw sql to the raw query-string.
Like so;

```
let query = sphinx
  .select(['id'])
  .from('myIndex')
  .orderBy([{ column: 'createdAt', order: 'desc' }])
  .boundingBox({ ne, sw })
  .match(text)
  
  // append offset, limit shorthand style as sphinx does not support offset ${offset}
  const result = await sphinx.raw(`${query.toQuery()} limit ${offset}, ${limit};SHOW META;`)
```

The ability to extend the qb with something like ´limitRaw(offset, limit) => `${offset}, ${limit}`´ would really help, but seems not possible with current restrictions. Or am I missing something or any other ideas?
